janus_server/src/config.rs

```diff
@@ -41,11 +41,51 @@ pub struct AggregatorConfig {
     pub listen_address: SocketAddr,
     /// The aggregator's database configuration.
     pub database: DbConfig,
-    /// Logging configuration
+    /// Logging configuration.
     #[serde(default)]
     pub logging_config: TraceConfiguration,
 }
 
+/// Non-secret configuration options for the Janus Aggregation Job Creator job.
+///
+/// # Examples
+///
+/// ```
+/// use janus_server::config::AggregationJobCreatorConfig;
+///
+/// let yaml_config = r#"
+/// ---
+/// database:
+///   url: "postgres://postgres:postgres@localhost:5432/postgres"
+/// logging_config: # logging_config is optional
+///   force_json_output: true
+/// tasks_update_frequency_secs: 3600
+/// aggregation_job_creation_interval_secs: 60
+/// min_aggregation_job_size: 100
+/// max_aggregation_job_size: 500
+/// "#;
+///
+/// let _decoded: AggregationJobCreatorConfig = serde_yaml::from_str(yaml_config).unwrap();
+/// ```
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct AggregationJobCreatorConfig {
+    /// Configuration for the database backend to connect to.
+    pub database: DbConfig,
+    /// Logging configuration.
+    #[serde(default)]
+    pub logging_config: TraceConfiguration,
+    /// How frequently we look for new tasks to start creating aggregation jobs for, in seconds.
+    pub tasks_update_frequency_secs: u64,
+    /// How frequently we attempt to create new aggregation jobs for each task, in seconds.
+    pub aggregation_job_creation_interval_secs: u64,
```

Comment:
Hmm -- I sorta disagree here. To keep the same configuration semantics, I would need to have `Duration` (de)serialize from an integer number of seconds. (why seconds specifically, and not some other unit?) Then I would want the in-memory field to be named e.g. `tasks_update_frequency` while the equivalent YAML field would be named `tasks_update_frequency_secs` (since we want to have some context on what the numeric value means) -- not impossible, but more complexity.

Since this structure is supposed to mirror a configuration file, I think it is OK to temporarily use a number instead of a `Duration`. Note that we do translate to a `Duration` before using these values in program logic -- see https://github.com/divviup/janus/blob/2290fe4deb394d5b1532965fe045b4bca6a40300/janus_server/src/bin/aggregation_job_creator.rs#L101-L104.

Reply:
I'm fine with that conclusion. It's not clear that spreading types whose purpose is encoding of network messages into config files is appropriate. Maybe what we want here is to move `message::{Time, Interval, Duration}` into `mod time` to decouple those types from the transport layer but I don't think that kind of code Feng Shui is urgent.